packages/gestalt/src/SideNavigationTopItem.js

```diff
@@ -176,30 +183,58 @@ const SideNavigationTopItemWithForwardRef: AbstractComponent<Props, HTMLLIElemen
         onTap={({ event, dangerouslyDisableOnNavigation }) => {
           setSelectedItemId(itemId);
           onClick?.({ event, dangerouslyDisableOnNavigation });
+
+          if (sideNavigationCollapsed) setOverlayPreview(false);
         }}
       >
         <Box
           color={itemColor}
-          paddingY={2}
+          paddingY={collapsed ? undefined : 2}
+          width={collapsed ? 44 : undefined}
           minHeight={44}
           rounding={2}
           display="flex"
+          justifyContent={collapsed ? 'center' : undefined}
           alignItems="center"
-          dangerouslySetInlineStyle={{
-            __style: {
-              paddingInlineStart: nestingMargin,
-              paddingInlineEnd: '16px',
-            },
-          }}
+          position="relative"
+          dangerouslySetInlineStyle={
+            collapsed
+              ? undefined
+              : {
+                  __style: {
+                    paddingInlineStart: nestingMargin,
+                    paddingInlineEnd: '16px',
+                  },
+                }
+          }
         >
-          <Flex gap={{ row: 2, column: 0 }} height="100%" width="100%">
+          {collapsed && icon && notificationAccessibilityLabel ? (
+            <Box
```

Comment:
I wonder if we could dry this into a Subcomponents <NotificationDot />

Reply:
we can but not now. please😄